modules/logger/parser/parser.go

```diff
@@ -164,13 +164,15 @@ func parseAndStoreTestOutput(
 				previousTestName = "summary"
 				logWriter.writeLog(logger, "summary", data)
 
+			case isResultLine(data):
+				// We ignore result lines, because that is handled specially below.
+
 			case previousTestName != "":
 				// Base case: roll up to the previous test line, if it exists.
 				// Handles case where terratest log has entries with newlines in them.
 				logWriter.writeLog(logger, previousTestName, data)
 
-			case !isResultLine(data):
-				// Result Lines are handled below
+			default:
```

Comment:
The previous logic didn't make sense, so this was switched to `default` which makes more sense.